scripts/alignmentfile_pairedness.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+import pysam
+import argparse
+import pathlib
+
+def parse_command() -> argparse.Namespace:
+    """Function to parse the command line input
+    Args:
+        None
+    Return:
+        Namespace: returns the args as a standard Namespace object
+    """
+    parser = argparse.ArgumentParser(description='Determine if input BAM,CRAM,SAM file is paried or single end')
+    parser.add_argument('--input_reads',
+        required=True,
+        help="Path to the BAM,CRAM,SAM file. If providing a CRAM, you must also provide an input_reference.")
+    parser.add_argument('--input_reference',
+        help="For CRAM only, provide the reference file used when making the input_reads")
+    parser.add_argument('--max_reads',
+        default=200_000,
+        help="The max number of reads to examine to make PAIRED/SINGLE determination. default=%default")
+    parser.add_argument('--threads',
```

Comment:
Where is this used in the script?

Reply:
Good point. I was supposed to use this in the line 67: `samfile = pysam.AlignmentFile(args.input_reads, 'r', reference_filename=args.input_reference)`. It can take threads: https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignmentFile

Reply:
Done